test/floodsub.spec.js

```diff
@@ -4,107 +4,113 @@
-
-    // subscribe and wait for subscription to be received in the other peer
-    floodsub2.subscribe(topic)
-    floodsub2.on(topic, checkMessage)
-    await pWaitFor(() => {
-      const subs = floodsub1.getSubscribers(topic)
-
-      return subs.length === 1
+    floodsub.subscribe(topic)
+    floodsub.on(topic, () => {
+      callCount++
     })
 
-    await floodsub1.publish(topic, message)
-    await pWaitFor(() => messageReceived === true)
+    // the message should not be in the cache
+    expect(floodsub.seenCache.has(key)).to.be.false()
+
+    // receive the message once
+    await floodsub._processRpc(peerStream.id.toB58String(), peerStream, rpc)
 
-    expect(floodsub2.seenCache.has.callCount).to.eql(2) // Put also calls .has
-    expect(floodsub2.seenCache.put.callCount).to.eql(1)
-    expect(floodsub2._publish.callCount).to.eql(1) // Forward message
+    // should have received the message
+    expect(callCount).to.equal(1)
 
-    const [msgProcessed] = floodsub2._processRpcMessage.getCall(0).args
+    // should be in the cache now
+    expect(floodsub.seenCache.has(key)).to.be.true()
 
-    // Force a second process for the message
-    await floodsub2._processRpcMessage(msgProcessed)
+    // receive the message multiple times
+    await floodsub._processRpc(peerStream.id.toB58String(), peerStream, rpc)
+    await floodsub._processRpc(peerStream.id.toB58String(), peerStream, rpc)
+    await floodsub._processRpc(peerStream.id.toB58String(), peerStream, rpc)
 
-    expect(floodsub2.seenCache.has.callCount).to.eql(3)
-    expect(floodsub2.seenCache.put.callCount).to.eql(1) // No new put
-    expect(floodsub2._publish.callCount).to.eql(1) // Not forwarded
+    // should only have emitted the message once
+    expect(callCount).to.equal(1)
   })
 
   it('forwards normalized messages on publish', async () => {
-    sinon.spy(floodsub1, '_forwardMessage')
-    sinon.spy(utils, 'randomSeqno')
-
-    await floodsub1.publish(topic, message)
-    expect(floodsub1._forwardMessage.callCount).to.eql(1)
-    const [messageToEmit] = floodsub1._forwardMessage.getCall(0).args
+    sinon.spy(floodsub, '_forwardMessage')
 
-    const computedSeqno = utils.randomSeqno.getCall(0).returnValue
-    utils.randomSeqno.restore()
-    sinon.stub(utils, 'randomSeqno').returns(computedSeqno)
+    await floodsub.publish(topic, message)
+    expect(floodsub._forwardMessage.callCount).to.equal(1)
+    const [messageToEmit] = floodsub._forwardMessage.getCall(0).args
 
     const expected = utils.normalizeInRpcMessage(
-      await floodsub1._buildMessage({
-        receivedFrom: peer1.peerId.toB58String(),
-        from: peer1.peerId.toB58String(),
+      await floodsub._buildMessage({
+        receivedFrom: floodsub.peerId.toB58String(),
         data: message,
         topicIDs: [topic]
       }))
 
     expect(messageToEmit).to.eql(expected)
   })
+
+  it.skip('does not send received message back to original sender', async () => {
+
```

Comment:
Needs test?

Reply:
Yeah, just writing it.  I pushed an update to see if CI was healthier with the libp2p rc, it is!

Reply:
Added now.